insights/specs/datasources/files_dirs_number_of.py

```diff
@@ -0,0 +1,35 @@
+"""
+Custom datasources to get dict of file or dir numbers
+"""
+import json
+import os
+from insights.core.context import HostContext
+from insights.core.exceptions import SkipComponent
+from insights.core.filters import get_filters
+from insights.core.plugins import datasource
+from insights.core.spec_factory import DatasourceProvider
+from insights.specs import Specs
+
+
+@datasource(HostContext)
+def files_dirs_number(broker):
+    """ Return a dict of file numbers from the spec filter """
+    filters = sorted(get_filters(Specs.files_dirs_number_filter))
+    result = {}
+    if filters:
+        for item in filters:
+            if not item.endswith("/"):
+                item = item + "/"
+            if os.path.exists(item):
+                files_number = 0
+                dirs_number = 0
+                result[item] = {}
+                for name in os.listdir(item):
+                    if os.path.isfile(item + name):
+                        files_number = files_number + 1
+                    else:
+                        dirs_number = dirs_number + 1
```

Comment:
To check for the dir type, what about making it more restrict by using `os.path.isdir()`? 

Like, for a file `crw-rw-rw-  1 root   wheel      0x400007c Feb 18 18:30 ttywc` , we would not want to include that as a dir:
```
>>> os.path.isfile("/dev/ttywc")
False
>>> os.path.isdir("/dev/ttywc")
False
```